Source/gendung.cpp

```diff
@@ -96,8 +96,10 @@ bool WillThemeRoomFit(int floor, int x, int y, int minSize, int maxSize, int *wi
 	int xCount = 0;
 	int yCount = 0;
 
-	// BUGFIX: change '&&' to '||' (fixed)
-	if (x > DMAXX - maxSize || y > DMAXY - maxSize) {
+	if (x > DMAXX - maxSize && y > DMAXY - maxSize) {
+		return false;
+	}
+	if (x > DMAXX - minSize || y > DMAXY - minSize) {

```

Comment:
Just to make sure: is this really an "and for max, but or for min"?

Reply:
Yes, rooms will apparently always be `maxSize` on either W or H, so if that isn't possible with out OOB the first one will do an early return. The room also has to be at least minSize in the other direction so if that isn't possible on either then we can return false as well.

It might be worth it to be more clever about it write it as:
```cpp
if (!(x + maxSize > DMAXX && y + minSize > DMAXY) && !(x + minSize > DMAXX && y + maxSize > DMAXY)) {
    return false;
}
```

Reply:
Thanks for confirming.